### PR TITLE
Fix bug in link-local unmarshalling

### PR DESCRIPTION
**Pull request by aboch**
Related to https://github.com/docker/docker/issues/28606

Signed-off-by: Alessandro Boch <aboch@docker.com>

### CODE REVIEW

**Comment by mavenugo** [comment]
LGTM.

am surprised that this day-1 issue was not caught till now. Better late than never.

File: endpoint_info.go [hunk 1]
```diff
@@ -114,10 +114,10 @@ func (epi *endpointInterface) UnmarshalJSON(b []byte) error {
...
 	}
 	if v, ok := epMap["llAddrs"]; ok {
-		list := v.([]string)
+		list := v.([]interface{})
 		epi.llAddrs = make([]*net.IPNet, 0, len(list))
 		for _, llS := range list {
-			ll, err := types.ParseCIDR(llS)
+			ll, err := types.ParseCIDR(llS.(string))
 			if err != nil {
 				return types.InternalErrorf("failed to decode endpoint interface link-local address (%s) after json unmarshal: %v", llS, err)
```
Comment (mavenugo) [inline endpoint_info.go]: Can you also fix here ?

**Comment by dongluochen** [comment]
lgtm

**Comment by mavenugo** [comment]
CI is timing out and is not related to this PR. merging

**Pull request merged**